### ISSUE TITLE
`assert_is_empty` should prefer array repeat syntax

### ISSUE THREAD

**Issue by xtqqczze** [rust-lang/rust-clippy#17674]
### Summary

`assert_is_empty` should prefer array repeat syntax when suggesting an empty array.

### Reproducer

Code:

```rust
pub fn issue17674(v: Vec<&str>) {
    assert!(v.is_empty());
}
```

Current output:

```rust
pub fn issue17674(v: Vec<&str>) {
    assert_eq!(v, [] as [&str; 0]);
}
```

Desired output:

```rust
pub fn issue17674(v: Vec<&str>) {
    assert_eq!(v, [""; 0]);
}
```

https://rust.godbolt.org/z/nKob1n1Wr

### Version

```text
clippy 0.1.100 (5db7f4be8a 2026-09-01)
```

**Comment by CommanderStorm** [comment]
🤔 

how about this instead:

```rust
assert_eq!(v.as_slice(), []);
```

no "hacky"/confusing zero width array cast...

**Comment by CommanderStorm** [comment]
noticed after reading your other issues that this seems more nuanced than I thought
- https://github.com/rust-lang/rust-clippy/issues/17676
- https://github.com/rust-lang/rust-clippy/issues/17673

**Comment by joshka** [comment]
I don't think this is a good suggestion. `[""; 0]` is a more confusing way of writing this as it requires you to understand that the `""` is there purely to make the type inference work, not as a value that matters. It also falls down a bunch where the type is something without a good default value to use. So it's more complex to read, more complex to make a rule, and doesn't simplify something that's a problem. I think the tradeoff on this is not worth it.

**Comment by xtqqczze** [comment]
One thing worth pointing out, is that the rule already has a special case for `String`:

```diff
- assert!(x.is_empty());
+ assert_eq!(x, "");
```

**Comment by DanielEScherzer** [comment]
> I don't think this is a good suggestion. `[""; 0]` is a more confusing way of writing this as it requires you to understand that the `""` is there purely to make the type inference work, not as a value that matters. It also falls down a bunch where the type is something without a good default value to use. So it's more complex to read, more complex to make a rule, and doesn't simplify something that's a problem. I think the tradeoff on this is not worth it.

I agree, and semantically `[""; 0]` means "an array of empty strings of length 0" rather than "an array of strings of length 0", so the change would also be a bit less clear
